### PR TITLE
chore: use a more structured YAML format

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It refactors the existing YAML file that configures the parallel branches in the CI pipeline. Instead of having an array of simple items, we are adding a more complex structure, trying to enrich the model for future enhacements.

## Why is it important?
We'd like to support more complex CI configurations, like multiple platforms, or different conditions to run test stages, and previous configuration was not supporting it.

The original use case is to run nightly builds with different tags than in the PR build.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Related issues

- Relates #652

## Follow-ups
We want to iterate through this PR to enhance the CI flow for this PR, maybe using the `beatsWhen` to identify events triggering the build.

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -251,13 +251,20 @@ pipeline {
...
-def checkTestSuite(Map parallelTasks = [:], String suite, String tags) {
-  def regexps = [ "^e2e/_suites/${suite}/.*", "^.ci/.*", "^cli/.*", "^e2e/.*\\.go" ]
-  if ("${FORCE_SKIP_GIT_CHECKS}" == "true" || isGitRegionMatch(patterns: regexps, shouldMatchAll: false)) {
-    log(level: 'INFO', text: "Adding ${suite}:${tags} test suite to the build execution")
-    parallelTasks["${suite}_${tags}"] = generateFunctionalTestStep(suite: "${suite}", tags: "${tags}")
-  } else {
-    log(level: 'WARN', text: "The ${suite}:${tags} test suite won't be executed because there are no modified files")
+def checkTestSuite(Map parallelTasks = [:], Map item = [:]) {
+  def suite = item.suite
+  def platforms = suite.platforms
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:256]: https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/detail/PR-654/7/pipeline/279

CI is failing with:
>No such field found: field java.lang.String platforms

Any guess?

Comment (v1v) [reply .ci/Jenkinsfile:256]: is it a list? or maybe  `item.platforms` similar to `item.scenarios`

Comment (mdelapenya) [reply .ci/Jenkinsfile:256]: It's exactly like `scenarios`: https://github.com/elastic/e2e-testing/pull/654/files#diff-b87fe24438639288bccba6a647eac526dec8e1755dcf1fc7959434a3ba95404aR4-R6

Comment (mdelapenya) [reply .ci/Jenkinsfile]: My bad... I've just seen it... It's `item.platforms` instead of `suite.platforms`

Comment (mdelapenya) [reply .ci/Jenkinsfile]: # #
image

**Comment by mdelapenya** [comment]
I'm going to merge this. It seems a good starting point to add more functionality on top of it. The simplest one related to supporting more platforms (arm?): it would be just a matter of adding a new entry in the platforms element in the `.e2e-tests.yaml` file:

```yaml
    platforms:
      - "ubuntu-18.04"
      - "new-supported-platform"
```

**Pull request merged**